### PR TITLE
Modify repo to use jsoncpp and Vulkan-LoaderAndValidationLayers as submodules

### PR DESCRIPTION
**Pull request by mark-lunarg**
Change to the root of your local VulkanTools repo
git fetch
git checkout master-submodules-cleanup
git submodule update --init --recursive
run update_external_sources.[bat|sh] (This runs the python script and calls the LVL script)

@cnorthrop , Karl left the Android build files in LVL untouched, but the VT version has been modified to at least build correctly.  We should talk about splitting the build files up into a VT subset and an LVL set at some point.

Note that most of the good stuff is in the root cmakefiles.txt file.  The majority of this PR is removing LVL stuff from the VT tree.

Right now this will pull in a github branch of LVL -- before pushing it'll be updated to a commit on Master.

### CODE REVIEW

**Comment by karl-lunarg** [comment]
@cnorthrop: To expand on what @mark-lunarg said, I modified the Android build files in the VT repo with eb8473b to continue to build the VT and LVL content.  I basically changed Android.mk to look for the LVL content in the submodule instead of VT, where it was removed as a result of this PR.

The question for you really is if you want to continue to build LVL content in the Android build of this VT repo.  If not, then it should be an easy operation to modify the files I touched in this commit to simply remove the LVL targets and make this build generate only VT targets.

**Comment by mikew-lunarg** [comment]
On Linux, there doesn't seem to be a VT/build/test directory being constructed.  After what I believe should be a full build...

```
mikew@spork:VulkanTools !455 0 $ find . -name tests
./tests
./submodules/Vulkan-LoaderAndValidationLayers/tests
./dbuild/submodules/Vulkan-LoaderAndValidationLayers/tests
./build/submodules/Vulkan-LoaderAndValidationLayers/tests
mikew@spork:VulkanTools !456 0 $ 
```

**Comment by karl-lunarg** [comment]
That's because there's nothing left in VulkanTools/test that needs building.  There's nothing left there but scripts.

VulkanTools/CMakeLists.txt does not call `add_directory(tests)`.

There is a VulkanTools/tests/CMakeLists.txt file that should be deleted since it is not called and is now obsolete.

That being said, the devsim_layer_test.sh file needs to be modified to cope with the new location of LVL components, namely the loader and vkjson_info.

I suppose I can take on these changes.

Can this devsim test be run against the mock ICD?  If so, we can add it to travis CI.  I'll give that a try.

**Comment by karl-lunarg** [comment]
Oh, and it looks like devsim_layer_test.sh assumes that it resides in the build tree, so we will have to add at least a copy step for this script.

**Comment by mark-lunarg** [comment]
Karl, I was already working on this -- the cmakelists in the tests dir did copy the scripts to the right spot, so I figured I'd leave that part.

**Comment by karl-lunarg** [comment]
yep

**Comment by mikew-lunarg** [comment]
That devsim test runs against my NV ICD.  Because the devsim config file
actually overrides every configurable parameter, it will (should) run on *any
*ICD.  It is currently set up to run on our internal CI.

**Comment by mikew-lunarg** [comment]
Thanks, the latest commits are able to successfully run the devsim test.

One issue: there seems to be a potential for unnecessary rebuilding.  Using this build recipe:
```
git clone git@github.com:LunarG/VulkanTools.git
cd VulkanTools
git checkout mikew_devsim130
# git rebase origin/master-submodules-cleanup
git submodule update --init --recursive
make -j
make -j
```
If the rebase is commented out (my current config, not using the new submodule stuff), the 2nd make simply returns:
> make: Nothing to be done for 'all'.

If the rebase is uncommented, the 2nd make does a substantial rebuild.

**Comment by mikew-lunarg** [comment]
I suspect the submodule restructure requires some changes to that recipe, so ignore that for now.

**Comment by karl-lunarg** [comment]
Sorry for the churn on the devsim test.  Turned out to be a path specification problem in the Travis config.

**Comment by karl-lunarg** [comment]
We should get some feedback/approvals from other people working in VulkanTools, like @davidlunarg and @joey-lunarg.  In particular, please build the PR branch and make sure that things like vktrace still build and work as you expect.

**Comment by cnorthrop** [comment]
This needs some work for Android yet.  I'm getting the following multiple times when I run `android-generate.sh`:

```
python3: can't open file '../../../submodules/Vulkan-LoaderAndValidationLayers/scripts/lvl_genvk.py': 
python3: can't open file '../../../submodules/Vulkan-LoaderAndValidationLayers/scripts/lvl_genvk.py': 
python3: can't open file '../../../submodules/Vulkan-LoaderAndValidationLayers/scripts/lvl_genvk.py': 
...
[Errno 2] No such file or directory
Traceback (most recent call last):
  File "../../../scripts/lvl_genvk.py", line 20, in <module>
    from reg import *
ImportError: No module named 'reg'
[Errno 2] No such file or directory
Traceback (most recent call last):
  File "../../../scripts/lvl_genvk.py", line 20, in <module>
    from reg import *
ImportError: No module named 'reg'
...
```
Note that running `build_all.sh` on Linux is a good way to check the build.

**Comment by cnorthrop** [comment]
@karl-lunarg 

> The question for you really is if you want to continue to build LVL content in the Android build of this VT repo.

I do want to be able to build CubeWithLayers (and maybe any other demos) as we use that content to verify vktrace and screenshot.  But I don't think we need to build the loader or validation layers.

File: layer_factory/CMakeLists.txt [hunk 1]
```diff
@@ -30,15 +30,17 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../layers)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../layers)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../layers)
 
-
 include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}
-    ${CMAKE_CURRENT_SOURCE_DIR}/../layers
-    ${CMAKE_CURRENT_SOURCE_DIR}/../loader
-    ${CMAKE_CURRENT_SOURCE_DIR}/../include/vulkan
+#    ${CMAKE_CURRENT_SOURCE_DIR}/../layers
```
Comment (mark-lunarg) [inline layer_factory/CMakeLists.txt]: Reminder to remove commented out lines here...

**Comment by karl-lunarg** [comment]
@cnorthrop 
> This needs some work for Android yet. I'm getting the following multiple times when I run android-generate.sh:
python3: can't open file '../../../submodules/Vulkan-LoaderAndValidationLayers/scripts/lvl_genvk.py': 
python3: can't open file '../../../submodules/Vulkan-LoaderAndValidationLayers/scripts/lvl_genvk.py': 
python3: can't open file '../../../submodules/Vulkan-LoaderAndValidationLayers/scripts/lvl_genvk.py': 

I had originally verified that the CI build worked for Android and didn't know about `build_all.sh`.

With the submodule properly updated, `build_all.sh` gets much further and didn't fail until it tried to build the demos.  I will fix this with a simple change to this script to find the demos in the new location.  With this fix, `build_all.sh` now completes successfully.

**Comment by karl-lunarg** [comment]
@cnorthrop 

> I do want to be able to build CubeWithLayers (and maybe any other demos) as we use that content to verify vktrace and screenshot. But I don't think we need to build the loader or validation layers.

OK, I didn't disable the building of anything because I wasn't sure what LVL content you were pulling from this VT build.   I'll leave things alone, but if you want to go back later and turn off the building of anything, you can do that after we merge this PR.  Actually, I'm not sure that there is much you can disable since you do use the layers for CubeWithLayers and you probably don't build the loader anyway.

**Comment by cnorthrop** [comment]
I pushed a commit that adds the submodule init to build_all.sh, similar to how update_external_sources.sh is doing it.  Things are building for me now for Android.

**Comment by mark-lunarg** [comment]
Rebased to pick up VT build fixes and combine some commits, incorporated review feedback, other fixes.

**Comment by mark-lunarg** [comment]
Pushed to Master.

**Pull request closed without merging**